sovrin_node/persistence/idr_cache.py

```diff
@@ -91,18 +94,28 @@ def close(self):
         self._keyValueStorage.close()
 
     def currentBatchCreated(self, stateRoot):
-        self.unCommitted[stateRoot] = OrderedDict(self.currentBatchOps)
+        self.unCommitted.append((stateRoot, OrderedDict(self.currentBatchOps)))
         self.currentBatchOps = []
 
-    def batchRejected(self, stateRoot=None):
-        if stateRoot:
-            self.unCommitted[stateRoot] = OrderedDict(self.currentBatchOps)
-        else:
-            self.currentBatchOps = []
+    def batchRejected(self):
+        # Batches are always rejected from end of `self.unCommitted`
+        self.currentBatchOps = []
+        self.unCommitted = self.unCommitted[:-1]
 
     def onBatchCommitted(self, stateRoot):
```

Comment:
Why do we assume in the API that batches are always rejected from the end of self.unCommitted and do not pass state into `batchRejected`, but pass the state explicitly into `onBatchCommitted` and check for assertion?
Shouldn't we either pass state into both methods and check assertions in both method (that the state is equal to the first batch's state in onBatchCommitted and is equal to the last batch's state in batchRejected), or don't pass it at all?